getting-started/basicapp/README.md

```diff
@@ -0,0 +1,362 @@
+commands below, naming the instance 'basicapp'.
+```
+INSTANCE_NAME=basicapp
+MACHINE_TYPE=db-n1-standard-1
+REGION=us-central1
+gcloud sql instances create $INSTANCE_NAME --tier=$MACHINE_TYPE \
+  --region=$REGION
+```
+
+To set the root password, use the cloud console or the command
+```
+gcloud sql users set-password root % --instance basicapp --password [PASSWORD]  
+```
+
+To see the connection details for your instance use the command
+```
+gcloud sql instances describe $INSTANCE_NAME
+```
+
+Look for the ```connectionName```, which as the form 
+```$PROJECT:$REGION:$INSTANCE_NAME```. Also, look for the IP address of the
+instance.
+
+Connect to the your Cloud SQL instance with the command in the cloud shell
+```
+INSTANCE_NAME=basicapp
+gcloud sql connect $INSTANCE_NAME --user=root
+```
+
+Execute the statements in [data/database_setup.sql](data/database_setup.sql).
+
+### Working with the Database in a Local Development Environment (Optional)
+
+To install the Cloud SQL proxy locally follow the steps in [Using Cloud SQL
+with Go](https://cloud.google.com/go/getting-started/using-cloud-sql), for
+example on Mac, use the commands
+
+```
+curl -o cloud_sql_proxy https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.386
+chmod +x cloud_sql_proxy
+```
+
+Start the proxy with the commands
+
+```
+PROJECT=[your project]
+REGION=[your region]
+INSTANCE_NAME=basicapp
+CONNECTION_NAME=$PROJECT:$REGION:$INSTANCE_NAME
+./cloud_sql_proxy -instances="$CONNECTION_NAME"=tcp:3306
+```
+
+Run the web app in a separate terminal
+```
+DBPASSWORD=[your password]
+DBUSER=proxyuser
+DBHOST=127.0.0.1
+DATABASE=messagesdb
+export MYSQL_CONNECTION: "$DBUSER:$DBPASSWORD@tcp($DBHOST:3306)/$DATABASE"
+go run cmd/basicwebapp/basicwebapp.go
+```
+
+You can now use the same curl commands as above to send your friend a message.
+
+## Packaging in a Docker Container
+The basicapp example can package your application in a Docker container based
+on the [golang image](https://hub.docker.com/_/golang/), with the commands
+```
+docker build -f build/docker/Dockerfile -t basicwebapp-image .
+docker run -itd --rm --name basicwebapp \
+  -p 8080:8080 \
+  --env MESSAGE_SERVICE=mock \
+  basicwebapp-image
+```
+
+This will run the container with the mock messaging service. We will describe
+how to configure the app to connect to Cloud SQL below.
+
+Check that the application is running OK use the same curl commands as above.
+
+Stop the app with the command
+```
+docker stop basicwebapp
+```
+
+Build the Docker container with cloud build:
+```
+gcloud builds submit --config build/cb-docker.yaml .
+```
+
+Run in Docker, test that it works, and stop it
+```
+gcloud auth configure-docker
+PROJECT_ID=[your project]
+docker run -itd --rm -p 8080:8080 \
+  --name basicwebapp \
+  --env MESSAGE_SERVICE=mock \
+  gcr.io/$PROJECT_ID/basicwebapp
+docker ps
+curl -I http://localhost:8080
+docker stop basicwebapp
+```
+
+## Deploy the app to Flex
```

Comment:
Can this run on Standard?

Reply:
No, the code needs to be changed to add an import to Standard.